### PR TITLE
String Reversal Program in Multiple Languages

### LINKED ISSUE #827: [Feature Request] Add some programs related to string
### Description
I propose adding a set of programs that demonstrate how to reverse a string in various programming languages. This will help users learn and compare different implementations.

### Solution
I would like to add the programs.

Please assign this issue to me with labels

- gssoc
- gssoc-ext
- hacktoberfest2024 

---

**Would you like to work on this feature?**  
- [x] Yes  
- [ ] No

### PR DESCRIPTION
**Pull request by riyaa060**
### Description
This pull request adds a program that demonstrates how to reverse a string in multiple programming languages, including:

- Java
- C++
- Python

Fixes #827

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/basic-dsa/Strings/Reversing a string.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
+---
+id: Reversing a string
```
Comment (ajay-dhangar) [inline docs/basic-dsa/Strings/Reversing a string.md]: replace `id: Reversing a string` with `id: reversing-a-string` for best practices

File: docs/basic-dsa/Strings/Reversing a string.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+The simplest way to reverse a string is by using Python's slicing feature.
+
+#### Example Code
+
+```python
+# Reversing a string using slicing
+original_string = "Hello, World!"
+reversed_string = original_string[::-1]
+print(reversed_string)  # Output: "!dlroW ,olleH"
+
```
Comment (ajay-dhangar) [inline docs/basic-dsa/Strings/Reversing a string.md]: close your code example with ```` ``` ````

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
update few changes based on suggestions

**Comment by riyaa060** [comment]
Done with the changes.

algo error

**Comment by ajay-dhangar** [comment]
If you want to learn more about our technology and improve your skills, I suggest you run our project on your local system and practice. Our project is open source, which makes it easy for beginners to learn the technology and improve their self-confidence. You can also add this experience to your resume.

**Review by ajay-dhangar — APPROVED** [review]
Done

**Comment by riyaa060** [comment]
Could you please add these labels too

- hacktoberfest2024 
- hacktoberfest-accepted

**Pull request merged**